Signal/src/view controllers/CallViewController.swift

```diff
@@ -205,6 +226,191 @@ class CallViewController: UIViewController {
+        textMessageButton = createButton(imageName:"logoSignal",
+                                         action:#selector(didPressTextMessage),
+                                         buttonSize:buttonSize)
+        muteButton = createButton(imageName:"mute-inactive",
+                                  action:#selector(didPressMute),
+                                  buttonSize:buttonSize)
+        speakerPhoneButton = createButton(imageName:"speaker-inactive",
+                                          action:#selector(didPressSpeakerphone),
+                                          buttonSize:buttonSize)
+        videoButton = createButton(imageName:"video-active",
+                                   action:#selector(didPressVideo),
+                                   buttonSize:buttonSize)
+        hangUpButton = createButton(imageName:"endcall",
+                                    action:#selector(didPressHangup),
+                                    buttonSize:buttonSize)
+
+        // A container for 3 rows of ongoing call controls.
+        ongoingCallView = UIView()
+        self.view.addSubview(ongoingCallView)
+        let ongoingCallRows = [
+            rowWithSubviews(subviews:[textMessageButton, videoButton],
+                            fixedHeight:buttonSize),
+            rowWithSubviews(subviews:[muteButton, speakerPhoneButton,],
+                            fixedHeight:buttonSize),
+            rowWithSubviews(subviews:[hangUpButton],
+                            fixedHeight:buttonSize),
+            ]
+        let ongoingCallRowSpacing = ScaleFromIPhone5To7Plus(20, 25)
+        var lastRow : UIView?
+        for row in ongoingCallRows {
+            ongoingCallView.addSubview(row)
+            row.autoPinWidthToSuperview()
+            if lastRow != nil {
+                row.autoPinEdge(.top, to:.bottom, of:lastRow!, withOffset:ongoingCallRowSpacing)
+            }
+            lastRow = row
+        }
+
+        ongoingCallView.setContentHuggingVerticalHigh()
+        ongoingCallRows.first!.autoPinEdge(toSuperviewEdge:.top)
+        ongoingCallRows.last!.autoPinEdge(toSuperviewEdge:.bottom)
+
+        // Incoming call controls
+        acceptIncomingButton = createButton(imageName:"call",
+                                            action:#selector(didPressAnswerCall),
+                                            buttonSize:buttonSize)
+        declineIncomingButton = createButton(imageName:"endcall",
+                                             action:#selector(didPressDeclineCall),
+                                             buttonSize:buttonSize)
+
+        incomingCallControlsRow = rowWithSubviews(subviews:[acceptIncomingButton, declineIncomingButton],
+                                                  fixedHeight:buttonSize)
+        self.view.addSubview(incomingCallControlsRow)
+    }
+
+    func createButton(imageName : String!, action : Selector!, buttonSize : CGFloat!) -> UIButton {
```

Comment:
Generally, prefer plain non-optional types whenever possible. As far as I can tell, there's no reason we'd ever want to pass any of these params as nil, so we'd want the compiler to error at the call site.

Unwrapping parameters in a function signature is not very common. The only valid use case I can think of is when implementing an objc API that didn't have nullability annotations, yet somehow you *know* the params will not be nil.

See http://stackoverflow.com/questions/24223120/in-swift-what-does-the-symbol-mean-in-a-function-signature